tests/unit/matrix/test_assembler.py

```diff
@@ -230,7 +231,69 @@ def but_it_provides_a_2D_columns_margin_for_an_MR_X_cube_result(
 
         columns_margin = assembler.columns_margin
 
-        SumSubtotals_.blocks.assert_called_once_with([[1, 2], [3, 4]], dimensions_)
+        SumSubtotals_.blocks.assert_called_once_with(
+            [[1, 2], [3, 4]], dimensions_, diff_cols_nan=True
+        )
+        _assemble_matrix_.assert_called_once_with(assembler, [[[1], [2]], [[3], [4]]])
+        assert columns_margin == [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
+
+    def it_provides_a_1D_columns_margin_proportion_for_a_CAT_X_cube_result(
+        self,
+        _rows_dimension_prop_,
+        dimension_,
+        _cube_result_matrix_prop_,
+        cube_result_matrix_,
+        _column_subtotals_prop_,
+        _column_order_prop_,
+        _assemble_vector_,
+    ):
+        _rows_dimension_prop_.return_value = dimension_
+        dimension_.dimension_type = DT.CAT
+        _cube_result_matrix_prop_.return_value = cube_result_matrix_
+        cube_result_matrix_.table_margin = 6.0
+        cube_result_matrix_.columns_margin = np.array([1, 2, 3])
+        _column_subtotals_prop_.return_value = [3, 5]
+        _column_order_prop_.return_value = [0, -2, 1, 2, -1]
+        _assemble_vector_.return_value = [1, 3, 2, 3, 5]
+        assembler = Assembler(None, None, None)
+
+        columns_margin = assembler.columns_margin_proportion
+
+        _assemble_vector_.assert_called_once_with(
```

Comment:
I had trouble with this, and makes me think there may be a better way to write this test (or the code underlying). `cube_result_matrix_.columns_margin` must be an `np.ndarray` instead of a list because it gets divided and `[1, 2, 3] / 6.0` gives error about not division not being defined on list & int. But then with the `ndarray` the `assert_called_once_with` struggles to compare.

Reply:
This is exactly how I would solve it, use `ANY` for the offending args and isolate those args for separate testing with `method_x_.call_args.args[i]`.

`.assert_called_once_with()` is a convenience matcher, just frequently saving us from doing this with every single call-arg, but when it won't work like in this (pretty rare) case, we need to do that work for ourselves. Looks fine to me, although of course a little more work to read, but whatever right?, sometimes code takes some work to interpret, our job is just to avoid _unnecessary_ work for the reader.